core/src/docs/rfcs/2758_merge_append_into_write.md

```diff
@@ -0,0 +1,79 @@
+- Proposal Name: `merge_append_into_write`
+- Start Date: 2023-08-02
+- RFC PR: [apache/incubator-opendal#2758](https://github.com/apache/incubator-opendal/pull/2758)
+- Tracking Issue: [apache/incubator-opendal#0000](https://github.com/apache/incubator-opendal/issues/0000)
```

Comment:
Does it need to use the 0000 non-exist issues link?

Reply:
Let's create a tracking issue since this RFC is likely to be accpeted.